include/deal.II/lac/read_write_vector.templates.h

```diff
@@ -404,6 +404,11 @@ namespace LinearAlgebra
                                   VectorOperation::values                         operation,
                                   std::shared_ptr<const CommunicationPatternBase> communication_pattern)
   {
+    // While the import does work with Trilinos 12.8.x, it fails with 12.4.x. To be safe,
+    // we disable it here. Note that it would be a useful case, as ReadWriteVector is
+    // supposed to replace ghosted vectors anyways.
+    AssertThrow(!trilinos_vec.has_ghost_elements(),
+                ExcMessage("Import() from TrilinosWrappers::MPI::Vector with ghost entries is not supported!"));
```

Comment:
Hm, but this seems wrong. Isn't `import()` supposed to get these values, doing whatever is necessary? So if these values are already available locally, shouldn't we be willing to just take them?

Reply:
We use the `import` function from Trilinos with the source being the distributed vector  ```Note that the SourceMap must have GIDs uniquely owned, each GID of the source map can occur only once.```

Reply:
What I'm saying is that the *intent* of the function is to get the necessary values out of the Trilinos vector into the read-write vector. How that is *implemented* is a different issue. So maybe we use the Trilinos `import` function when the vector has no ghost elements, and if there are ghost elements, then we just copy them. (I'd be ok with throwing an exception if the vector does have ghost elements, but they don't match the ones we need.)

Reply:
I see. If we want to support that we need to use the `export` function instead of the `import` function. That will work as long as the indices in ReadWriteVector are unique (which they should be since the ReadWriteVector is serial).

Reply:
@bangerth The problem is that this function allows you to import whatever you want from the Trilinos vector so there is no reason this has to align with the owned/ghosted elements. Note that I added a function that does precisely what you want here: https://github.com/dealii/dealii/pull/5015

Reply:
Almost. #5015 also sets the index set of the RW vector to exactly those elements that the Trilinos vector locally stores; it then copies over these elements. Here, you already have an index set given, and you want the corresponding values from the Trilinos vector. 

As I said above, I see in essence three cases:
* Trilinos vector has no ghost elements. In that case, the existing `import()` call is correct and sufficient.
* Trilinos vector has ghost elements and the set of locally owned and ghost elements of the Trilinos vector is a superset of the indices the RW vector wants. In that case, we can just copy over. This is the case where you currently assert out.
* Trilinos vector has ghost elements, but not the right ones. In that case, one would have to use `import()` again, but I suspect that this case actually rarely happens in practice. I'd be ok to not treat this case right away but assert out for the moment.

Reply:
I understand. But I don't see a good reason to use a ReadWriteVector in combination with ghosted vectors and a subset of indices. If you use ReadWriteVector in the future, you shouldn't be using ghosted vectors anyways. For the shim I am writing (route ghosted vectors through the ReadWriteVector implementation of DataOut, etc. as we discussed at the hackathon), the indexsets will be the same, so #5015 is enough.

That said, if you want to implement and test this new functionality, go ahead. I don't have time to work on it right now.

Reply:
I see your point with not using ghosted vectors anyway.

OK to merge then.